internal/kafka/internal/config/dataplane_cluster_config.go

```diff
@@ -306,24 +296,18 @@ func (c *DataplaneClusterConfig) IsReadyDataPlaneClustersReconcileEnabled() bool
 	return c.EnableReadyDataPlaneClustersReconcile
 }
 
-// DefaultComputeMachineType returns the default Compute Machine Type for the
-// given `cloudProviderID`. If `cloudProviderID` is not a known cloud provider
-// the empty string is returned.
-func (c *DataplaneClusterConfig) DefaultComputeMachineType(cloudProviderID cloudproviders.CloudProviderID) string {
-	switch cloudProviderID {
-	case cloudproviders.AWS:
-		return c.AWSComputeMachineType
-	case cloudproviders.GCP:
-		return c.GCPComputeMachineType
-	default:
-		return ""
+// DefaultComputeMachineType returns the Compute Machine Type config for the
+// given `cloudProviderID`. If `cloudProviderID` is not a known cloud provider return an error.
+func (c *DataplaneClusterConfig) DefaultComputeMachineType(cloudProviderID cloudproviders.CloudProviderID) (MachineTypeConfig, error) {
```

Comment:
Can we rename the method to ComputeMachineTypeConfig

Reply:
or to DefaultComputeMachineTypeConfig

Reply:
I'll go with `DefaultComputeMachineTypeConfig` Thanks for catching this

Reply:
modified